Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,130 +7,145 @@
-    expectedFiles << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd/eee.fff.ggg/basefile.txt");//adding a folder name containing dots
-    expectedFiles << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd/eee.fff.ggg/basefile1.txt");
-    expectedFiles << tempPath.absoluteFilePath("subfolder3/basefile.txt");
-    expectedFiles << tempPath.absoluteFilePath("subfolder3/aaa/basefile.txt");
-    expectedFiles << tempPath.absoluteFilePath("subfolder3/aaa/bbb/basefile.txt");
-    expectedFiles << tempPath.absoluteFilePath("subfolder3/aaa/bbb/ccc/basefile.txt");
-    expectedFiles << tempPath.absoluteFilePath("rootfile1.txt");
-
-    for (const QString& expect : expectedFiles)
+    TEST_F(AssetScannerUnitTest, AssetScanner_ScanMultipleFolders_ExpectedFilesAndFoldersFound)
     {
-        UNIT_TEST_EXPECT_TRUE(CreateDummyFile(expect));
-    }
-
-    // but we're going to not watch subfolder3 recursively, so... remove these files (even though they're on disk)
-    // if this causes a failure it means that its ignoring the "do not recurse" flag:
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/basefile.txt"));
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/basefile.txt"));
-    expectedFiles.remove(tempPath.absoluteFilePath("subfolder3/aaa/bbb/ccc/basefile.txt"));
-
-    QSet<QString> expectedFolders;
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd");
-    expectedFolders << tempPath.absoluteFilePath("subfolder2/aaa/bbb/ccc/ddd/eee.fff.ggg");
-
-
-    PlatformConfiguration config;
-    AZStd::vector<AssetBuilderSDK::PlatformInfo> platforms;
-    config.PopulatePlatformsForScanFolder(platforms);
-    //                                               PATH               DisplayName  PortKey outputfolder  root recurse  platforms
-    config.AddScanFolder(ScanFolderInfo(tempPath.absolutePath(),         "temp",       "ap1",    true,  false, platforms));  // note:  "Recurse" set to false.
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder1"), "",           "ap2",   false,  true,  platforms));
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder2"), "",           "ap3",   false,  true,  platforms));
-    config.AddScanFolder(ScanFolderInfo(tempPath.filePath("subfolder3"), "",           "ap4",   false,  false, platforms)); // note:  "Recurse" set to false.
-    AssetScanner scanner(&config);
-
-    QList<AssetFileInfo> filesFound;
-    QList<AssetFileInfo> foldersFound;
-
-    bool doneScan = false;
-
-    connect(&scanner, &AssetScanner::FilesFound, this, [&filesFound](QSet<AssetFileInfo> fileList)
+        using namespace AssetProcessor;
+        QSharedPointer<QCoreApplication> m_qApp;
```

Comment:
Why do you need shared ownership of the `QCoreApplication` instance? It is a singleton, and is always available via `QCoreApplication::instance()`.

Reply:
it's not initialized by default, QCoreApplication::instance() returns nullptr if I don't create it. The shared pointer lets me make sure I don't leave the QCoreApplication instance active after this test runs.

Reply:
If you don't need shared ownership, but want automatic cleanup, use a `AZStd::unique_ptr` or `QScopedPtr` instead.

Reply:
I've changed it to unique_ptr